Add Interop 2024 Goals and Roadmap (#19)
web-platform-tests/interop-mobile-testing: README.md

# interop-mobile-testing
Mobile Testing Investigation in Interop

## 2024 Investigation

### Scope

> Do the work so that by Interop 2025 the dashboard has a mobile view, and the mobile view is considered as reliable as desktop.

Reviewer: ```suggestion
Do the work so that by Interop 2025 the dashboard has a mobile view, with at least "experimental" versions of browsers, and the mobile view is considered as reliable as desktop.
```

### Roadmap

> 1. Formulate and implement a plan for getting Mobile Safari Results onto the dashboard

Reviewer: ```suggestion
1. Formulate and implement a plan for getting iOS WebKit Results onto the dashboard
```

Per discussion last year and in the last meeting, the current intention is to get WebKitTestRunner running on iOS Simulator.

Apple only ships stable and beta releases of Mobile Safari, thus if the goal is to have something such that we can show as our "experimental" label, merely running Mobile Safari does not suffice.

Mobile Safari also has no control over viewport size (and this is a feature basically required by testing and nothing else—web developers typically want to test as the browser actually ships to users).

> 1. For each mobile browser, investigate and catalouge test failures relative to desktop

Reviewer: ```suggestion
1. For each mobile browser, investigate and catalogue test failures relative to desktop
```

> 1. Reduce the number of unexpected mobile test failures relative to desktop to 0

Reviewer: ```suggestion
1. Reduce the number of unexplained mobile test failures relative to desktop to 0
```

Maybe that clarifies things for @past?

> 1. Reduce the number of unexpected mobile test failures relative to desktop to 0

Reviewer 2: Resolved in https://github.com/web-platform-tests/interop-mobile-testing/pull/19#discussion_r1479719599
